core/src/main/java/com/graphhopper/reader/dem/AbstractTiffElevationProvider.java

```diff
@@ -37,91 +32,122 @@
  *
  * @author Robin Boldt
  */
-public abstract class AbstractTiffElevationProvider implements ElevationProvider {
-    final Logger logger = LoggerFactory.getLogger(getClass());
-    final Map<String, HeightTile> cacheData = new HashMap<String, HeightTile>();
-    protected Downloader downloader;
-    File cacheDir;
-    String baseUrl;
-    private Directory dir;
-    private DAType daType = DAType.MMAP;
-    boolean calcMean = false;
-    boolean autoRemoveTemporary = true;
-    long sleep = 2000;
-
-    public AbstractTiffElevationProvider(String baseUrl, String cacheDir, String downloaderName) {
+public abstract class AbstractTiffElevationProvider extends AbstractElevationProvider {
+    private final Map<String, HeightTile> cacheData = new HashMap<String, HeightTile>();
+    final double precision = 1e7;
+
+    private final int WIDTH;
+    private final int HEIGHT;
+
+    // Degrees of latitude covered by this tile
+    final int LAT_DEGREE;
+    // Degrees of longitude covered by this tile
+    final int LON_DEGREE;
+
+    public AbstractTiffElevationProvider(String baseUrl, String cacheDir, String downloaderName, int width, int height, int latDegree, int lonDegree) {
+        super(cacheDir);
         this.baseUrl = baseUrl;
-        this.cacheDir = new File(cacheDir);
-        downloader = new Downloader(downloaderName).setTimeout(10000);
+        this.downloader = new Downloader(downloaderName).setTimeout(10000);
+        this.WIDTH = width;
+        this.HEIGHT = height;
+        this.LAT_DEGREE = latDegree;
+        this.LON_DEGREE = lonDegree;
     }
 
     @Override
-    public void setCalcMean(boolean eleCalcMean) {
-        calcMean = eleCalcMean;
-    }
+    public void release() {
+        cacheData.clear();
 
-    void setSleep(long sleep) {
-        this.sleep = sleep;
+        // for memory mapped type we create temporary unpacked files which should be removed
+        if (autoRemoveTemporary && dir != null)
+            dir.clear();
     }
 
     /**
-     * Creating temporary files can take a long time as we need to unpack tiff as well as to fill
-     * our DataAccess object, so this option can be used to disable the default clear mechanism via
-     * specifying 'false'.
+     * Return true if the coordinates are outside of the supported area
      */
-    public void setAutoRemoveTemporaryFiles(boolean autoRemoveTemporary) {
-        this.autoRemoveTemporary = autoRemoveTemporary;
-    }
+    abstract boolean outsideSupportedArea(double lat, double lon);
```

Comment:
I know this is not a setter but do we need Java code style here: isOutsideSupportedArea?

Reply:
Makes sense :). I changed the method name.